content/reference/custom_interactions.md

```diff
@@ -0,0 +1,221 @@
++++
+title = 'Custom Interactions'
+weight = 6
++++
+
+Buttons, Modals, Select Menus, Ephemeral Messages, and more!
+
+<!--more-->
+
+{{% notice warning %}}
+
+Use of interactions within YAGPDB is an advanced topic. You will need a thorough understanding of YAGPDB's scripting
+language before learning interactions.
+
+We have a comprehensive learning course located [here](https://learn.yagpdb.xyz) for you to get started.
+
+{{% /notice %}}
+
+## The Basics
+
+Interactions within Discord allow server members to use alternative, built-in features to trigger bots to take action
+other than messages or reactions. These features include builtin buttons, dropdown selection menus, or submitting a
+modal (basically a pop-up form). Within custom commands it is possible to both create and customize these new
+interactive features, but respond to them as well, opening up new possibilities for ephemeral message responses, modals,
+and more within YAGPDB custom templating script.
+
+### Definitions
+
+**Interaction** - A user engaging with YAGPDB through one of Discord's builtin features: Clicking a button, Making a
+selection with a select menu, or Submitting a modal.
+
+**Response** - YAGPDB is required to send an initial interaction response after receiving an *interaction*. If it does
+not do this, the user triggering the interaction will see a "This application did not respond" error. The bot cannot
+respond to an interaction more than once. YAGPDB may only respond to an interaction in one of the following ways:
+Sending a message, sending a modal, or updating the message which the *interaction* was triggered on.
+
+**Followup** - Since YAGPDB may only *respond* to an *interaction* once, it is subsequently required to send an interaction
+followup if it still needs to respond to the interaction. These followups can be sent up to 15 minutes after the initial
+interaction, and you can send as many as you want. YAGPDB may only send a followup in one of the following ways: Sending
+a message, editing a response or followup, or getting a response or followup message.
+
+**Interactive Elements** - Elements users can interact with to send *interactions*, i.e. buttons, menus, and modals.
+
+**Message Components** - *Interactive Elements* which can be attached to YAGPDB's Discord messages, i.e. buttons and menus.
```

Comment:
I understand these are the terms that the Discord API uses, but I would rather inline these terms into their use sites. So instead of saying "Creating message components" I would much rather see "Creating buttons and menus"; this seems more amenable to skimming (and, in general, more appropriate for an audience that does not work with the Discord API directly.)

Reply:
I'm trying to get people familiar with the idea of message components for a few reasons:

1. A component of a message makes it clear that these are just parts of any regular message, not just interaction responses. During prototyping a few users struggled to understand how to create buttons or menus in the first place.
2. The CC trigger type is a "message component" type.

Reply:
I think this is fine if we write the other reference pages in a similar style, where appropriate.
Otherwise I agree with Joe -- in-depth explanations belong, in my opinion, to learn.yagpdb.

For now I propose we keep this as-is and then once we migrate the learning resources over to Hugo as well we move details over there. What do you think?